catalogue/webapp/components/SearchContext/SearchContext.js

```diff
@@ -0,0 +1,106 @@
+// @flow
+import { type Node, createContext, useState, useContext } from 'react';
+import Router from 'next/router';
+import { worksUrl } from '@weco/common/services/catalogue/urls';
+
+export type CatalogueQuery = {|
+  query: string,
+  page: number,
+  workType: string[],
+  itemsLocationsLocationType: string[],
+  queryType: ?string,
+|};
+
+type ContextProps = {|
+  ...CatalogueQuery,
+  setQuery: (value: string) => void,
+  setPage: (value: number) => void,
+  setWorkType: (value: string[]) => void,
+  setItemsLocationsLocationType: (value: string[]) => void,
+  setQueryType: (value: ?string) => void,
+|};
+
+const defaultState: CatalogueQuery = {
+  query: '',
+  page: 1,
+  workType: ['k', 'q'],
+  itemsLocationsLocationType: ['iiif-image'],
+  queryType: null,
+};
+
+const SearchContext = createContext<ContextProps>({
+  ...defaultState,
+  setQuery: () => {},
```

Comment:
Should these functions be updating the state or am I missing something?

Reply:
When creating a context, this is just the default thing to return if there isn't a provider.
Maybe this should just be `null`, but this makes it super `null`-checky.